python/ml4ir/applications/classification/model/classification_model.py

```diff
@@ -103,6 +93,56 @@ def evaluate(
             self.logger.info(f"Evaluation Results written at: {logs_dir}")
         return global_metrics, grouped_metrics, metrics_dict
 
+    @staticmethod
+    def get_chunks_from_df(dataframe, size):
+        """
+        Given a pd.DataFrame, creates batches of size `size`.
+        """
+        return [dataframe.iloc[pos:pos + size] for pos in range(0, len(dataframe), size)]
+
+    def calculate_metric_on_batch(self, metric, predictions, batch_size, group_name=None, group_key=None):
+        """
+        Given a metric and a dataframe with `predictions`, it iterates the dataframe in
+        using `batch_size` and updates the metric. Once the dataframe is fully iterated,
+        the score of the metric is returned as a dictionary containing
+        {"group_name": group_name,
+         "group_key": group_key,
+         "metric": metric.name,
+         "value": metric.result().numpy(),
+         "size": predictions.shape[0]
+        }
+        When group_name and group_key are not provided, they are exluded from the dict.
+
+        Parameters
+        ----------
+        metric : an instance of tf.keras.metrics
+        predictions : pd.DataFrame
+        batch_size: int
+        group_name : str, optional
+            The name to be used in the returned dictionary
+        group_key : str, optional
+            A key to be used in the returned dictionary
+
+        Returns
+        -------
+        A dictionary with the metric scores for the given data.
+        """
+        label_name = self.feature_config.get_label()['name']
+        output_name = self.output_name
+        metric.reset_states()
+        for chunk in self.get_chunks_from_df(predictions, batch_size):
+            metric.update_state(tf.constant(chunk[label_name].values.tolist(), dtype=tf.float32),
```

Comment:
this is important, it gives a guge speedup as there is not tracing needed when using tensors.